android/app/src/main/java/org/fossasia/openevent/viewmodels/SessionDetailActivityViewModel.java

```diff
@@ -0,0 +1,77 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.SharedPreferencesUtil;
+
+import io.realm.RealmChangeListener;
+
+public class SessionDetailActivityViewModel extends ViewModel {
+
+    public static final String BY_ID = "id";
+    public static final String BY_NAME = "name";
+
+    private RealmDataRepository realmRepo;
+    private MutableLiveData<Session> sessionByIdLiveData;
+    private MutableLiveData<Session> sessionByNameLiveData;
+    private Session sessionById;
+    private Session sessionByName;
+
+    public SessionDetailActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<Session> getSessionById(int id, String loadedFlag) {
+        if (sessionByIdLiveData == null) {
+            sessionByIdLiveData = new MutableLiveData<>();
+            sessionById = realmRepo.getSession(id);
+            sessionById.addChangeListener((RealmChangeListener<Session>) loadedSession -> {
+                if (!loadedSession.isValid())
```

Comment:
@iamareebjamal If i use a RealmWrapper i won't be able to fetch the Session that it wraps, inside the ViewModel and hence i would have to do these checks in the Activity. Is that fine ?  otherwise using MutableLiveData is better.

Reply:
MutableLiveData sounds good